### DISCUSSION TITLE
socket to metrics

### DISCUSSION THREAD

**Discussion by taikula-8** [vectordotdev/vector#21204] [Q&A]
Hello, i have a question for you. 

How to convert data received by a socket to a metric using transforms？

The received data is as follows：
{"host":"127.0.0.1","message":"{\"name\":\"node_memory_Active_anon_bytes\",\"help\":\"Memory information field Active_anon_bytes.\",\"type\":1,\"metric\":[{\"gauge\":{\"value\":4231172096}}]}","port":56348,"source_type":"socket","timestamp":"2024-09-03T12:15:09.118397820Z"}

I want to convert this json data into a metric. I checked the help documentation for transform and found that it doesn't seem to work well.

**Comment by jszwedko** [comment]
I think you could use `log_to_metric` to generate a metric from that log event.
